pkg/bundle/client_test.go

```diff
@@ -0,0 +1,170 @@
+		bundle := givenPackageBundle(api.PackageBundleStateInactive)
+		mockClient.EXPECT().Get(ctx, gomock.Any(), gomock.AssignableToTypeOf(&pbc))
+
+		active, err := bundleClient.IsActive(ctx, bundle)
+
+		assert.True(t, active)
+		assert.Nil(t, err)
+	})
+
+	t.Run("error on failed get", func(t *testing.T) {
+		mockClient := newMockClient(t)
+		bundleClient := NewPackageBundleClient(mockClient)
+		bundle := givenPackageBundle(api.PackageBundleStateInactive)
+		mockClient.EXPECT().Get(ctx, gomock.Any(), gomock.AssignableToTypeOf(&pbc)).Return(fmt.Errorf("failed get"))
+
+		_, err := bundleClient.IsActive(ctx, bundle)
+
+		assert.NotNil(t, err)
+	})
+
+	t.Run("fails on wrong namespace", func(t *testing.T) {
+		mockClient := newMockClient(t)
+		bundleClient := NewPackageBundleClient(mockClient)
+		bundle := &api.PackageBundle{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: "Wrong-Name",
+			},
+			Status: api.PackageBundleStatus{
+				State: api.PackageBundleStateActive,
+			},
+		}
+		mockClient.EXPECT().Get(ctx, gomock.Any(), gomock.AssignableToTypeOf(&pbc))
+
+		active, err := bundleClient.IsActive(ctx, bundle)
+
+		assert.False(t, active)
+		assert.Nil(t, err)
+	})
+
+	t.Run("fails on wrong name", func(t *testing.T) {
+		mockClient := newMockClient(t)
+		bundleClient := NewPackageBundleClient(mockClient)
+		bundle := &api.PackageBundle{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "non-empty",
+				Namespace: api.PackageNamespace,
+			},
+			Status: api.PackageBundleStatus{
+				State: api.PackageBundleStateActive,
+			},
+		}
+		mockClient.EXPECT().Get(ctx, gomock.Any(), gomock.AssignableToTypeOf(&pbc))
+
+		active, err := bundleClient.IsActive(ctx, bundle)
+
+		assert.False(t, active)
+		assert.Nil(t, err)
+	})
+}
+
+func TestBundleClient_GetActiveBundle(t *testing.T) {
+	t.Parallel()
+
+	ctx := context.Background()
+	pbc := newMockPackageBundleController()
+
+	t.Run("golden path", func(t *testing.T) {
+		mockClient := newMockClient(t)
+		bundleClient := NewPackageBundleClient(mockClient)
+		mockBundle := newMockBundle()
+
+		mockClient.EXPECT().Get(ctx, gomock.Any(), gomock.AssignableToTypeOf(&pbc))
+		mockClient.EXPECT().Get(ctx, gomock.Any(), gomock.AssignableToTypeOf(&mockBundle)).SetArg(2, mockBundle)
+
+		bundle, err := bundleClient.GetActiveBundle(ctx)
+
+		assert.Equal(t, bundle.Name, "test-name")
+		assert.Nil(t, err)
+	})
+}
+
+func TestBundleClient_GetActiveBundleNamespacedName(t *testing.T) {
+	t.Parallel()
+
+	ctx := context.Background()
+	pbc := newMockPackageBundleController()
+
+	t.Run("golden path", func(t *testing.T) {
+		mockClient := newMockClient(t)
+		bundleClient := NewPackageBundleClient(mockClient)
+		mockClient.EXPECT().Get(ctx, gomock.Any(), gomock.AssignableToTypeOf(&pbc))
+
+		namespacedNames, err := bundleClient.GetActiveBundleNamespacedName(ctx)
+
+		assert.Equal(t, api.PackageNamespace, namespacedNames.Namespace)
+		assert.Equal(t, "", namespacedNames.Name)
+		assert.Nil(t, err)
+	})
+}
+
+// Helpers
```

Comment:
My minor nit on these helper names, for tests, I normally call these something like `givenMockClient` so they don't show up when someone is trying to remember what the name of some new function. Also given follows the testing given/when/then pattern. We are putting tests in the same package as our code and the danger with doing that is someone will accidentally use test code in a production implementation.